lex/gen.go

```diff
@@ -576,6 +576,11 @@ func (s *TypeSchema) WriteRPC(w io.Writer, typename string) error {
 		return fmt.Errorf("can only generate RPC for Query or Procedure (got %s)", s.Type)
 	}
 
+	if fname != "ServerRefreshSession" {
+		pf("\tc.Mux.RLock()\n")
+		pf("\tdefer c.Mux.RUnlock()\n")
+	}
```

Comment:
I don't love this. Really feels like any locking should happen in the client, and only when needed (aka, when actually updating auth info). Should be possible to do concurrent requests from the same client.

Reply:
This is a readlock, you can do concurrent requests, it just makes sure you're not trying to make a request while the client object is being modified.